src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -199,6 +199,26 @@ public ReadOnlyDataBlobKey(Utf8String name, byte[] data, int alignment)
             public override bool Equals(object obj) => obj is ReadOnlyDataBlobKey && Equals((ReadOnlyDataBlobKey)obj);
             public override int GetHashCode() => Name.GetHashCode();
         }
+
+        protected struct FieldRvaKey : IEquatable<FieldRvaKey>
+        {
+            public readonly int Rva;
+            public readonly EcmaModule Module;
+
+            public FieldRvaKey(int rva, EcmaModule module)
+            {
+                Rva = rva;
+                Module = module;
+            }
+
+            public bool Equals(FieldRvaKey other) => Rva == other.Rva && Module.Equals(other.Module);
+            public override bool Equals(object obj) => obj is FieldRvaKey && Equals((FieldRvaKey)obj);
+            public override int GetHashCode()
+            {
+                int hashCode = Rva * 0x5498341 + 0x832424;
```

Comment:
Should we use the TypeHashingAlgorithm.ComputeSignatureVariableHashCode here to compute the hash here? I am not sure if that makes sense, but I was wondering where the formula came from and I've found the same formula in there.

Reply:
This formula is a copy-paste from a few lines above. This is just used for the internal hashtable book-keeping of RVA blobs, to ensure we don't created duplicate nodes for the same RVA blob. 

The TypeHashingAlgorithms are intended for version resilient hashes written by the R2R compiler, and consumed by the runtime typically.

Reply:
Ok. It would be nice to have the hashing formulas in one place for all the usages in crossgen2 (there are four places now using the same formula), but it is not in scope of this PR.